lib/kv-redis.js

```diff
@@ -97,17 +97,24 @@ RedisKeyValueConnector.prototype.ping = function(cb) {
  *
  * @param {String} command The Redis command to execute.
  * @param {Array} args List of options for the given command.
- * @callback {Function} callback
+ * @options {Object} options (optional) This is currently not used but included
+ * to match the signature in the loopback-datasource-juggler
+ * @callback {Function} cb Callback invoked on redis command
  * @param {Error} err Error object.
  * @param {*} result The result of the command execution.
  *
- * @header RedisKeyValueConnector.prototype.execute(command, args, cb)
+ * @header RedisKeyValueConnector.prototype.execute(command, args, options, cb)
  */
 RedisKeyValueConnector.prototype.execute = function(command, args, cb) {
```

Comment:
Changing the signature to add in the `options` argument seemed to break many tests that live outside this repository and expect a 3 argument signature. Since the `options` argument is not used anywhere in the code I decided that it might be cleaner to just check to see if the method was invoked with 4 arguments and then re-assign the 4th argument to the callback if it is a function.

Reply:
> Changing the signature to add in the `options` argument seemed to break many tests that live outside this repository and expect a 3 argument signature.

Can you please provide more details? Which tests are failing? What was the particular implementation you have chosen? Even with a 4-arg signature, it's important to preserve backwards compatibility and support the 3-arg variant too.